### PR TITLE
Fix #812 (incorrect mac paths)

### LINKED ISSUE #812: Some mounted docker paths are referenced as /var instead of /var/private in Mac Serina
Example: the 90th conformance test. When executing this actual command is executed without the quiet flag, the following happens:
```
$ /Users/th10/checkouts/cwltool/venv/bin/cwltool --outdir=/var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpy3_of70y ../common-workflow-language/v1.0/v1.0/cat3-tool.cwl ../common-workflow-language/v1.0/v1.0/file-literal.yml
/Users/th10/checkouts/cwltool/venv/bin/cwltool 1.0.20180703025338
Resolved '../common-workflow-language/v1.0/v1.0/cat3-tool.cwl' to 'file:///Users/th10/checkouts/common-workflow-language/v1.0/v1.0/cat3-tool.cwl'
[job cat3-tool.cwl] /private/tmp/docker_tmpldgg2u5f$ docker \
    run \
    -i \
    --volume=/private/tmp/docker_tmpldgg2u5f:/private/var/spool/cwl:rw \
    --volume=/private/var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpeuf451sm:/tmp:rw \
    --volume=/var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpeuf451sm/tmpjsh0kge1:/private/var/lib/cwl/stg15c81a0c-99ab-4202-b300-953e76cfc7d8/21bdddd8-b055-41d7-b6c1-e57713e784c1:rw \
    --workdir=/private/var/spool/cwl \
    --read-only=true \
    --log-driver=none \
    --user=16442:1312 \
    --rm \
    --env=TMPDIR=/tmp \
    --env=HOME=/private/var/spool/cwl \
    debian:stretch-slim \
    cat \
    /private/var/lib/cwl/stg15c81a0c-99ab-4202-b300-953e76cfc7d8/21bdddd8-b055-41d7-b6c1-e57713e784c1 > /private/tmp/docker_tmpldgg2u5f/output.txt
docker: Error response from daemon: Mounts denied:
The path /var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpeuf451sm/tmpjsh0kge1
is not shared from OS X and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> File Sharing.
See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
.
ERRO[0000] error waiting for container: context canceled
[job cat3-tool.cwl] completed permanentFail
{
    "output_file": {
        "location": "file:///var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpy3_of70y/output.txt",
        "basename": "output.txt",
        "class": "File",
        "checksum": "sha1$da39a3ee5e6b4b0d3255bfef95601890afd80709",
        "size": 0,
        "path": "/var/folders/5p/p6z8hhy174q_ctt3pn7mx5_c000h1t/T/tmpy3_of70y/output.txt"
    }
}
Final process status is permanentFail
```

As you can see from the 3rd `--volume`, `/var` is not being expanded to `/var/private`

### PR DESCRIPTION
**Pull request by ThomasHickman**
This correctly interprets mounted paths in mac and sets the TMPDIR correctly inside docker, so all the conformance tests pass on mac machines

### CODE REVIEW

File: cwltool/docker.py [hunk 1]
```diff
@@ -309,7 +309,7 @@ def create_runtime(self, env, runtimeContext):
         if runtimeContext.rm_container:
             runtime.append(u"--rm")
 
-        runtime.append(u"--env=TMPDIR=/tmp")
+        runtime.append(u"--env=TMPDIR=%s" % self.builder.tmpdir)
```
Comment (mr-c) [inline cwltool/docker.py]: This should be a path within the container; `self.tmpdir` is mounted to `/tmp` at https://github.com/common-workflow-language/cwltool/pull/814/files#diff-616bb24f4b86bb222b6d286980bd2373L274

Comment (ThomasHickman) [reply cwltool/docker.py]: Ah ok, `runtime.tmpdir` must be set incorrectly then

Comment (ThomasHickman) [reply cwltool/docker.py]: I think the problem is actually at https://github.com/common-workflow-language/cwltool/blob/977962eca49fc126b839e8965e3f08240027bd0a/cwltool/process.py#L626 - paths that are passed into arguments to docker are resolved in the host environment, so `/tmp` gets translated to `/private/tmp` which doesn't match up with what we've specified as `TMPDIR`. I'll commit the changes that fix this

**Comment by mr-c** [comment]
Thank you @ThomasHickman !

**Pull request merged**